app/models/service_provider.rb

```diff
@@ -78,11 +78,11 @@ def redirect_uris=(uris)
     super uris.select(&:present?)
   end
 
-  # @return [ServiceProviderCertificate]
-  def certificate
-    @certificate ||= begin
-      if saml_client_cert
-        ServiceProviderCertificate.new(OpenSSL::X509::Certificate.new(saml_client_cert))
+  # @return [Array<ServiceProviderCertificate>]
+  def certificates
+    @certificates ||= [*certs, *saml_client_cert].map do |cert|
+      if cert
```

Comment:
Could this ever be `nil`?

Reply:
Oh, with the `*`... I guess not lol, great catch

Reply:
> Oh, with the `*`... I guess not lol, great catch

I was originally going to comment about whether the array should be `[*certs, saml_client_cert]`, but I see the `nil` handling is quite handy 😄